### ISSUE TITLE
`heroku pg:backups capture` stuck on "Pending"

### ISSUE THREAD

**Issue by briankung** [heroku/heroku#1459]
![screen shot 2015-03-19 at 10 20 32 am](https://cloud.githubusercontent.com/assets/2836167/6733684/afcdffca-ce21-11e4-80ab-4736fceb9992.png)

I can't figure out what's going on with this command because the `heroku pg:backups info` is returning a heroku internal client error (https://github.com/heroku/heroku/issues/1458) and the `--verbose` flag itself seems to be broken (https://github.com/heroku/heroku/issues/1457).

Any tips would be appreciated.

**Comment by diwu1989** [comment]
I also see pg:backups capture command being stuck very often

**Comment by yvesvanbroekhoven** [comment]
+1

**Comment by smaziano** [comment]
Getting the same error also... I had to cancel it to move on last build... that it's cleary not a good practice.
+1

**Comment by dickeyxxx** [comment]
any ideas here @uhoh-itsmaciek?

**Comment by dickeyxxx** [comment]
I'm wondering if it is related to the exception handling we added here

**Comment by uhoh-itsmaciek** [comment]
I don't think so--I think a lot of these were because until earlier today, we didn't prioritize manual backups over scheduled backups, so things could be pending for a long time with a surge of scheduled backups (e.g., as happens nightly). In any case, I don't think there's any aspect of the CLI that could be causing issues here: it just reports things from the backend, and the retries we put in would only show things in `Pending` for a few minutes. If this happens again, I'd file a support ticket at help.heroku.com unless it's clearly a CLI problem.

**Comment by Proffard** [comment]
+1
I'm getting stuck at "Pending" when I try to restore backup from url manually.

**Comment by mliu7** [comment]
+1
I'd been using pgbackups for years without any issues but when switching over to pg:backups today I'm also stuck on Pending. I tried this on multiple apps and databases from multiple machines, all running the updated heroku toolbelt and all are stuck on Pending at the moment.

**Comment by mliu7** [comment]
As a follow up, all of the backups eventually started and completed successfully. It appeared they were queued by about 15-20 minutes. If you could somehow get back to the old behavior where backups start right away that'd be awesome.

**Comment by dogweather** [comment]
Thanks for the info @mliu7 

My backups manual restore also seems to hang on pending, and I haven't found any docs or indication of what my expectations should be. It seems to be non-functional, but I'll wait it out.

**Comment by Jaco-Pretorius** [comment]
Same thing happening here, it's intermittent - both for capture and restore.

**Comment by uhoh-itsmaciek** [comment]
For anyone affected by this: we are aware of this issue and we're working on it. In the meantime, you may sometimes see backups in `Pending` for a long time. It's not a client issue, so there's not much we can do in this project. I apologize for the inconvenience.

**Comment by tef** [comment]
To give a little more information about what is going wrong and what we're doing to fix it:

– We're having throughput and congestion problems, so some jobs are being delayed a considerable amount. We've put in a temporary fix to prioritize interactive backups over scheduled backups, but we're working on overhauling the scheduling and worker system.

– The stalling transfers and delays are encouraging users to start multiple backups to see if one gets started, which then puts extra congestion on our system.

– Some jobs are hanging, due to some deep and nasty locking issues, and we're investigating but it's been very hard to replicate to diagnose the problem—The individual parts of the system do not show the issue. The hanging jobs then take up capacity.

As we've migrated more and more people across to the new backup system, we've encountered numerous edge cases, despite a long period of internal testing and beta access. Atop of this, the problems above have a knock-on effect, we end up with more contention and more stalled jobs.

We're aware of the problems and pain this is causing and we are doing our best to fix these issues (and others) as quickly as possible. To reiterate what @uhoh-itsmaciek said, these aren't issues with the toolbelt, but with our internal tools. 

The earlier system caused a lot of pain for us behind the scenes and, in hindsight, we were a little too eager to push out the new system. We're also going over our processes to ensure that these categories of problems do not happen so easily again.

In summary: I'm sorry—I know the system is causing you a lot of pain, and I'm working on fixing this as quickly as possible.

**Comment by tef** [comment]
Good news everyone, I hope. We found the cause of the hanging backup/restore jobs, and we've shipped the fix. 

However, there is still a backlog of jobs to process, and of course, some of our users are running extra jobs to mitigate for the availability issues. We expect things to smooth out over the weekend and backups to continue as normal.

Please open a ticket if your backup is hanging without completion, or held in pending for a long time.

**Comment by tef** [comment]
As a follow up: We haven't been encountering stalling backups, or backups taking a significant time to start, or missing nightly backups any more. The latter was caused by some very slow API calls, which has now been fixed.

Again: I'm sorry for the problems and please open a ticket if you're encountering problems with the new system.

**Comment by billba** [comment]
I'm experiencing this same problem when doing a "heroku pg:backups restore". I did this successfully a week ago, but now it's stalling on "pending".

**Comment by bishtawi** [comment]
@tef, I am encountering pg backups hanging on the "pending" state today.

**Comment by tef** [comment]
Hi, an API change knocked workers offline, we found the fault it's been operational for the last 30-40 minutes.

Please open a ticket with heroku directly if you are having a problem with the service, not a github issue for the toolbelt.
